### ISSUE TITLE
PW2 Utilization figures

### ISSUE THREAD

**Issue by nitro182** [mihailescu2m/powerwall_monitor#6]
So, not really an issue but maybe an ask for help. So I'm in the process of changing around the dashboard (which I am happy to share anything and everything with you in case you want to use it)

Next to the Powerwall in/out, solar/consumption figures I've added a Powerwall Utilization and a Powerwall Efficiency graph. Efficiency works fine, here's the code if you want to look at using/testing/improving it :)

SELECT (f/t)*100 FROM (SELECT integral("from_pw")  / 1000 / 3600   AS f, integral("to_pw")  / 1000 / 3600  AS t FROM "autogen"."http" WHERE  $timeFilter tz('Australia/Sydney'))

Now when we look at Utilisation, it also works fine but only when looking at a day of data, i.e. today, yesterday etc. The figures break when you start to look at a month of data which is where ideally I'd want to view it at:

SELECT f / 27 * 100 FROM (SELECT integral("from_pw")  / 1000 / 3600  AS f FROM "autogen"."http" WHERE  $timeFilter tz('Australia/Sydney'))

Basically we want to look at the from_pw data/number of days/PW capacity - in my case I have two of them so 27 kWh.

Do you know how this query could be optimised to calculate the current time period selected in the graph to calculate the utilisation?

Cheers,
Adam.

**Comment by mihailescu2m** [comment]
You can try and use "from_pw" from "kwh" retention policy (`select from_pw as f from kwh.http ` i think), it's already converted into kwh and don't need to use integral()/1000/3600 :)

**Comment by nitro182** [comment]
Ok cool, let me keep playing. Thanks!
